src/pymor/models/iosys.py

```diff
@@ -700,6 +701,132 @@ def __mul__(self, other):
                         T=self.T, initial_data=initial_data, time_stepper=time_stepper, num_values=self.num_values,
                         solver_options=self.solver_options)
 
+    def impulse_resp(self, mu=None, return_solution=False):
+        """Compute impulse response from all inputs.
+
+        Parameters
+        ----------
+        mu
+            |Parameter values| for which to solve.
+        return_solution
+            If `True`, the model solution for the given |parameter values| `mu` is returned.
+
+        Returns
+        -------
+        y
+            Impulse response as a 3D |NumPy array| where
+            `y.shape[0]` is the number of snapshots,
+            `y.shape[1]` is the number of outputs, and
+            `y.shape[2]` is the number of inputs.
+        Xs
+            The tuple of solution |VectorArrays| for every input.
+            Returned only when `return_solution` is `True`.
+        """
+        assert self.T is not None
+
+        if not isinstance(mu, Mu):
+            mu = self.parameters.parse(mu)
+
+        # solution computation
+        common_solver_opts = dict(
+            operator=-self.A,
+            mass=None if isinstance(self.E, IdentityOperator) else self.E,
+            initial_time=0,
+            end_time=self.T,
+            num_values=self.num_values,
+        )
+        if self.sampling_time == 0:
+            Xs0 = self.E.apply_inverse(self.B.as_range_array(mu=mu), mu=mu)
+            Xs = tuple(
+                self.time_stepper.solve(
+                    rhs=None,
+                    initial_data=X0,
+                    mu=mu,
+                    **common_solver_opts,
+                )
+                for X0 in Xs0
+            )
+        else:
+            rhs = LinearInputOperator(self.B)
+            Xs = []
+            for i in range(self.dim_input):
+                def input_i(t):
+                    if t == 0:
+                        e_i = np.zeros(self.dim_input)
+                        e_i[i] = 1/self.sampling_time
+                        return e_i
+                    return np.zeros(self.dim_input)
+                Xs.append(
+                    self.time_stepper.solve(
+                        rhs=rhs,
+                        initial_data=self.solution_space.zeros(1),
+                        mu=mu.with_(input=GenericFunction(input_i, shape_range=(self.dim_input,))),
+                        **common_solver_opts,
+                    )
+                )
+            Xs = tuple(Xs)
+
+        # output computation
```

Comment:
If only the output is requested, you use much more memory than needed by keeping all trajectories in memory.

Reply:
It would be nice if the time-steppers returned a generator. Without that, I don't see how to avoid keeping the entire trajectory.